lib/pbench/server/indexing_tarballs.py

```diff
@@ -319,15 +319,10 @@ def sighup_handler(*args):
                         idxctx.logger.info("Starting {} (size {:d})", tb, size)
                         dataset = None
```

Comment:
What's the scope of `dataset`?  It appears to be assigned at line 327 and read at line 344...if that's correct, then we don't need this initialization here, especially if we move the read from the `else` to the inside of `try` block.

Reply:
I thought I'd answered this before, but apparently not. The initialization to `None` (which is now at line 320) ensures that the reference in the `finally` way down at line 428 is safe regardless of which exception paths we might take. That's where I advance the dataset state to `INDEXED` on success or `QUARANTINED` on failure, assuming we found the dataset in the first place. (And of course we always should.)